### PR TITLE
simplify some if statements

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/metadata/ffmpeg_handler.cc [hunk 1]
```diff
@@ -357,13 +357,11 @@ std::unique_ptr<IOHandler> FfmpegHandler::serveContent(const std::shared_ptr<Cds
...
 
-    const auto cacheEnabled = config->getBoolOption(CFG_SERVER_EXTOPTS_FFMPEGTHUMBNAILER_CACHE_DIR_ENABLED);
-
-    if (cacheEnabled) {
-        if (auto data = readThumbnailCacheFile(item->getLocation())) {
-            log_debug("Returning cached thumbnail for file: {}", item->getLocation().c_str());
-            return std::make_unique<MemIOHandler>(data->data(), data->size());
-        }
+    auto cacheEnabled = config->getBoolOption(CFG_SERVER_EXTOPTS_FFMPEGTHUMBNAILER_CACHE_DIR_ENABLED);
+    auto data = readThumbnailCacheFile(item->getLocation());
```
Comment (KarlStraussberger) [inline src/metadata/ffmpeg_handler.cc:361]: You cannot call read... if cache is not enabled. Costs time and it may cause exception on cache file access!

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]
ffmpeg changes are breaking

File: src/metadata/ffmpeg_handler.cc (hunk 1 above)
Comment (neheb) [reply src/metadata/ffmpeg_handler.cc:361]: I noticed that too. I'll fix.

Comment (neheb) [reply src/metadata/ffmpeg_handler.cc]: Fixed.

**Pull request merged**